UI/cypress/e2e/results.js

```diff
@@ -109,4 +113,31 @@ Then("display all correct tuition partners that provide the selected subjects in
 
 Then("display all correct tuition partners in any location", () => {
     cy.get('[data-testid="results-list-item"]').should("exist")
+});
+
+Then("the ‘clear filters’ button is displayed at the bottom of the filters section", () => {
+    cy.get('[data-testid="clear-all-filters"]').should("exist").should('contain.text', 'Clear all filters');
+});
+
+Then("no subject should be shown as selected", () => {
+    cy.get('.app-results-filter').contains('text', 'checked').should('not.exist');
+});
+
+Then("all subject filters are collapsed", () => {
+    stages.forEach(element => 
+        cy.get(`[data-testid=${kebabCase(element)}]`).
+            should('have.class', 'js-closed')
+)});
+
+Then("results are updated after filters are cleared", () => {
+    let countOfElements = 0;
+    cy.get('[data-testid="results-list-item"]').then($elements => {
+    countOfElements = $elements.length;
+    cy.get('[data-testid="clear-all-filters"]').click();
+    cy.get('[data-testid="results-list-item"]').should('have.length.greaterThan', countOfElements)
+    })
+});
+
+Then("the postcode search parameter remains", () => {
+    cy.get('[data-testid="postcode-input-box"]').should('not.be.null');
```

Comment:
This test doesn't fail if the code behind is altered to null out the postcode. The default for the input box is an empty string which is not null. Best to change this to look for the exact postcode entered to prove that it doesn't alter the postcode

Reply:
Changed